src/EditorFeatures/CSharp/DecompiledSource/CSharpDecompiledSourceService.cs

```diff
@@ -103,14 +103,26 @@ public static async Task<Document> FormatDocumentAsync(Document document, Syntax
 
             var fullTypeName = new FullTypeName(fullName);
 
-            // Try to decompile; if an exception is thrown the caller will handle it
-            var text = decompiler.DecompileTypeAsString(fullTypeName);
+            try
+            {
+                var text = decompiler.DecompileTypeAsString(fullTypeName);
+
+                text += "#if false // " + CSharpEditorResources.Decompilation_log + Environment.NewLine;
+                text += logger.ToString();
+                text += "#endif" + Environment.NewLine;
 
-            text += "#if false // " + CSharpEditorResources.Decompilation_log + Environment.NewLine;
-            text += logger.ToString();
-            text += "#endif" + Environment.NewLine;
+                return document.WithText(SourceText.From(text));
+            }
+            catch (NotSupportedException)
+            {
+                // IL Spy throws for "Decompiling types that are not part of the main module is not supported."
+            }
+            catch (InvalidOperationException)
```

Comment:
Should this be filtering on the exception message? I recognize that's potentially sketchy but I worry that catching such broad exceptions is also problematic.

Reply:
The rest of the code here is string concatenation and ToStringing a string builder. I can isolate the catch to just the ILSpy call if that makes you feel better?